src/Models/Model.php

```diff
@@ -328,6 +328,10 @@ public function isPropertyTypeValid( string $key, $value ) : bool {
 				return is_bool( $value );
 			case 'array':
 				return is_array( $value );
+			case 'float':
+				return is_float( $value );
+			case 'numeric':
+				return is_numeric( $value );
```

Comment:
I'd prefer this to be a union, as I don't want to include numeric strings at this time. And let's call it "number" as not to be confusing with "numeric".
```suggestion
			case 'number':
				return is_int( $value ) || is_float( $value );
```